build-tools/scripts/jenkinsfile

```diff
@@ -0,0 +1,380 @@
+
+def botLabel = 'xamarin-android-master-pipeline-eval'
+
+def XADir = "xamarin-android"
+
+def publishBuildFilePaths = null
+def publishTestFilePaths = null
+
+def MSBUILD_AUTOPROVISION_ARGS="/p:AutoProvision=True /p:AutoProvisionUsesSudo=True /p:IgnoreMaxMonoVersion=False"
+
+def buildStatusProcessed = false
+def buildStatusUploaded = false
+
+def isPr = false                // Default to CI
+
+def processBuildStatus() {
+    def status = 0
+
+    try {
+        echo "processing build status"
+        sh "make package-build-status CONFIGURATION=${env.BuildFlavor}"
+    } catch (error) {
+        echo "ERROR : NON-FATAL : processBuildStatus: Unexpected error: ${error}"
+        status = 1
+    }
+
+    return status
+}
+
+def publishPackages(filePaths) {
+    def status = 0
+    try {
+         // Note: The following function is provided by the Azure Blob Jenkins plugin
+         azureUpload(storageCredentialId: "${env.StorageCredentialId}",
+                                storageType: "blobstorage",
+                                containerName: "${env.ContainerName}",
+                                virtualPath: "${env.StorageVirtualPath}",
+                                filesPath: "${filePaths}",
+                                allowAnonymousAccess: true,
+                                pubAccessible: true,
+                                doNotWaitForPreviousBuild: true,
+                                uploadArtifactsOnlyIfSuccessful: true)
+    } catch (error) {
+        echo "ERROR : publishPackages: Unexpected error: ${error}"
+        status = 1
+    }
+
+    return status
+}
+
+pipeline {
+    
+    agent {
+        label "${botLabel}"
+    }
+
+    options {
+        timestamps()
+    }
+
+    stages {
+        stage('init') {
+            steps {
+                script {
+                    // Note: PR plugin environment variable settings available here: https://wiki.jenkins.io/display/JENKINS/GitHub+pull+request+builder+plugin
+                    isPr = env.ghprbActualCommit != null
+                    def branch = env.GIT_BRANCH
+                    def commit = isPr ? env.ghprbActualCommit : env.GIT_COMMIT
+                    def buildType = isPr ? 'PR' : 'CI'
+
+                    echo "Job: ${env.JOB_BASE_NAME}"
+                    echo "Branch: ${branch}"
+                    echo "Commit: ${commit}"
+                    echo "Build type: ${buildType}"
+                    if (isPr) {
+                        echo "PR id: ${env.ghprbPullId}"
+                        echo "PR link: ${env.ghprbPullLink}"
+                    }
+
+                    publishBuildFilePaths = "xamarin-android/xamarin.android-oss*.zip,xamarin-android/bin/${env.BuildFlavor}/bundle-*.zip,xamarin-android/bin/Build*/Xamarin.Android.Sdk*.vsix,xamarin-android/prepare-image-dependencies.sh,xamarin-android/build-status*,xamarin-android/xa-build-status*";
+                    echo "publishBuildFilePaths: ${publishBuildFilePaths}"
```

Comment:
Why echo this message as part of `init` when they don't exist yet?

Reply:
Addressed with commit https://github.com/xamarin/xamarin-android/commit/2b2579740a63971efd2fc891bf4143bddc88ca6f
I've moved the logging for these settings at the point they are used